src/embind/embind.js

```diff
@@ -663,7 +668,12 @@ var LibraryEmbind = {
             HEAPU32[ptr >> 2] = length;
 
             if (stdStringIsUTF8 && valueIsOfTypeString) {
-                Module['stringToUTF8'](value, ptr + 4, length, false);
+                try {
+                    Module['stringToUTF8'](value, ptr + 4, length + 1, false);
```

Comment:
Why the +1? The docs for stringToUTF8 refer to a parameter named skipNullTerminator but I don't see the name skipNullTerminator referenced anywhere in preamble.js.